tests/acceptance/30_generic_methods/unsafe/user_fullname.cf

```diff
@@ -0,0 +1,84 @@
+
+#######################################################
+#
+# Test checking user_fullname
+#
+#######################################################
+
+bundle common acc_path
+{
+  vars:
+    "root" string => getenv("NCF_TESTS_ACCEPTANCE", 1024);
+}
+
+body common control
+{
+      inputs => { "${acc_path.root}/default.cf.sub", "${acc_path.root}/default_ncf.cf.sub", "@{ncf_inputs.default_files}" };
+      bundlesequence  => { configuration, default("${this.promise_filename}") };
+      version => "1.0";
+}
+
+#######################################################
+
+bundle agent init
+{
+  vars:
+    "user1"       string => "user1";
+    "user2"       string => "user2";
+    "fullname2"    string => "fullname2";
+}
+
+#######################################################
+
+bundle agent test
+{
+  methods:
+    "ph1" usebundle => user_fullname("${init.user1}", "${init.fullname2}");
+    "ph2" usebundle => user_present("${init.user2}");
+    "ph3" usebundle => user_fullname("${init.user2}", "${init.fullname2}");
+}
+
+#######################################################
+
+bundle agent check
+{
+  vars:
+    pass1::
+      "user_line1" string => execresult("/bin/grep '^${init.user1}:' /etc/passwd", "useshell");
+      "user_line2" string => execresult("/bin/grep '^${init.user2}:' /etc/passwd", "useshell");
+
+  classes:
+    pass1::
+      "user1_ok" expression => strcmp("", "${user_line1}");
+      "user2_ok" expression => regcmp("^(${init.user2}):x:([0-9]+):\2:(${init.fullname2}):\/home\/\1:", "${user_line2}");
```

Comment:
I don't understand the logic here :(
on pass1, you first delete user (userdel), then check if user is there. It should fail.